packages/jupyter-ai/jupyter_ai/tests/completions/test_handlers.py

```diff
@@ -88,8 +89,36 @@ async def test_handle_request(inline_handler):
     assert suggestions[0].insertText == "Test response"
 
 
+async def test_handle_request_with_spurious_fragments(inline_handler):
+    inline_handler.create_llm_chain(
+        MockProvider,
+        {
+            "model_id": "model",
+            "responses": ["```python\nTest python code\n```"],
```

Comment:
What if there is a trailing new line after the triple backtick? Will the test still passe? Maybe worth to paramterise it?